mysql/azuremysql/azuremysql.go

```diff
@@ -48,6 +61,45 @@ type Params struct {
 	TraceOpts []ocsql.TraceOption
 }
 
+// URLOpener opens Azure MySQL URLs
+// like "azuremysql://user:password@mydemoserver.mysql.database.azure.com:3306/mydb".
+type URLOpener struct {
+	// CertSource specifies how the opener will obtain authentication information.
+	// CertSource must not be nil.
+	CertSource CertPoolProvider
```

Comment:
If it can't be `nil`, then it shouldn't be `nil` in `RegisterMySQL`. I addressed this in #1685 by using the lazy-creds opener pattern we've been using across the project.